### PR TITLE
Reduce KubeVirt e2e test flakiness

### PR DESCRIPTION
**Pull request by davidvossel**
We're encountering some issues with flaky tests on the kubevirt platform. There are two issues here that this PR addresses.

1. We need the ability to alter the guest memory used for the worker nodes in the prow configs. We're seeing in 4.11 that some of the default OCP components can't start on the worker nodes due to lack of memory. This is addressed by the new `e2e.kubevirt-node-memory` e2e test cli option
2. Running OVNKubernetes as the network overlay in the worker nodes is flaky at the moment. I'm investigating this, but for now I think we should switch back to the e2e default which uses the SDN.

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -238,7 +240,7 @@ func (o *options) DefaultClusterOptions() core.CreateOptions {
 			ServicePublishingStrategy: kubevirt.IngressServicePublishingStrategy,
 			ContainerDiskImage:        o.configurableClusterOptions.KubeVirtContainerDiskImage,
 			Cores:                     2,
-			Memory:                    "4Gi",
+			Memory:                    o.configurableClusterOptions.KubeVirtNodeMemory,
```
Comment (alvaroaleman) [inline test/e2e/e2e_test.go:243]: Why make this configurable rather than just hardcoding a working size? This now requires users to know what the right size it

Comment (davidvossel) [reply test/e2e/e2e_test.go:243]: There's still the cli default of "4Gi".

I was hoping to make the memory size configurable because it allows us to adapt our prow configs without needing to do this dance between modifying the hypershift codebase and then modifying prow.

For example, today the e2e tests use 5 replicas of 4Gi worker nodes. This configuration is occasionally failing because in 4.11 we have some pods that fail to schedule on the workers due to 4Gi being too little memory per worker.

I think 2 replica's of 8Gi will more reliably fit all the pods. I'd like to make that kind of change just in the prow config instead of having to change a hardcoded value in hypershift's e2e test which will cause the kubevirt e2e tests to fail until i can update the prow config to modify the replica count.

basically, memory is closely related to replica count for us, so it would be helpful to expose this via the cli just like replica count.

**Review by alvaroaleman — APPROVED** [review]
